src/Table/write_ipac_table/write_ipac_table.cxx

```diff
@@ -8,92 +8,95 @@
 
 // JTODO Reduce code duplication in loops.
 
+std::vector<size_t> tablator::Table::get_effective_array_sizes(
+        const std::map<size_t, Data_Type> &datatypes_for_writing) const {
+    std::vector<size_t> array_sizes;
+    array_sizes.emplace_back(1);  // null column
+    for (size_t i = 1; i < columns.size(); ++i) {
+        auto column = columns[i];
+        Data_Type active_datatype = get_datatype_for_writing(datatypes_for_writing, i);
+        size_t effective_array_size =
+                (active_datatype == Data_Type::CHAR) ? 1 : column.array_size;
+        array_sizes.emplace_back(effective_array_size);
+    }
+    return array_sizes;
+}
+
+/**********************************************************/
+
 void tablator::Table::write_ipac_table(
-        std::ostream &os, const std::map<size_t, Data_Type> &datatypes_for_writing) const {
-    std::vector<size_t> ipac_column_widths = get_column_width();
+        std::ostream &os,
+        const std::map<size_t, Data_Type> &datatypes_for_writing) const {
+    std::vector<size_t> ipac_column_widths = get_column_widths();
     write_ipac_table_header(os);
+
+    std::vector<size_t> effective_num_arrays =
```

Comment:
Already done---you must not have been seeing the very latest.

Reply:
I am commenting in commit order, because the initial giant clang-format commit makes the overall diff basically unreadable. So there may be more of these sorts of comments - sorry!